javaslang-gwt/src/test/java/client/GwtTestCollections.java

```diff
@@ -0,0 +1,65 @@
+package client;
+
+import com.google.gwt.junit.client.GWTTestCase;
+import javaslang.Function1;
+import javaslang.Tuple;
+import javaslang.Tuple1;
+import javaslang.collection.*;
+
+public class GwtTestCollections extends GWTTestCase {
+
+    @Override
+    public String getModuleName() {
+        return "TestModule";
+    }
+
+    public void testTuple() {
+        Tuple1<Integer> t = Tuple.of(1);
+        assertEquals((int) t._1(), 1);
+    }
+
+    private void applyCollection(Function1<char[], Traversable<Character>> factory) {
+        Traversable<Character> traversable = factory.apply(new char[] {'a', 'b', 'c'});
+        assertEquals(traversable.count(i -> i != 'b'), 2);
+    }
+
+    public void testCompileArray() {
+        applyCollection(Array::ofAll);
+    }
+
+    public void testCompileBitSet() {
+        applyCollection(BitSet::ofAll);
+    }
+
+    public void testCompileCharSeq() {
+        applyCollection(chars -> CharSeq.of(new String(chars)));
```

Comment:
In this case test fails with `ClassCastException`. And I don't know what the problem. I changed this to `CharSeq.ofAll(Iterator.ofAll(chars))` (like other your comments)